packages/picasso.js/src/core/component/component-factory.js

```diff
@@ -551,11 +559,15 @@ function componentFactory(definition, context = {}) {
 
     setUpEmitter(instanceContext, emitter, config);
     setUpEmitter(definitionContext, emitter, definition);
+
+    instanceContext.isVisible = () => isVisible;
   };
 
   fn.mounted = () => mounted(element);
 
   fn.unmount = () => {
+    const isVisible = false;
```

Comment:
I would move `isVisible` declaration to the scope of `componentFactory` function and just toggle the variable in mount/unmount